src/libraries/Common/src/System/Net/Http/Http2/Hpack/HeaderField.cs

```diff
@@ -12,40 +11,23 @@ namespace System.Net.Http.HPack
         // http://httpwg.org/specs/rfc7541.html#rfc.section.4.1
         public const int RfcOverhead = 32;
 
-        public HeaderField(ReadOnlySpan<byte> name, ReadOnlySpan<byte> value)
+        public HeaderField(string name, string value)
         {
             Debug.Assert(name.Length > 0);
 
-            // TODO: We're allocating here on every new table entry.
-            // That means a poorly-behaved server could cause us to allocate repeatedly.
-            // We should revisit our allocation strategy here so we don't need to allocate per entry
-            // and we have a cap to how much allocation can happen per dynamic table
-            // (without limiting the number of table entries a server can provide within the table size limit).
-            Name = new byte[name.Length];
-            name.CopyTo(Name);
-
-            Value = new byte[value.Length];
-            value.CopyTo(Value);
+            Name = name;
+            Value = value;
```

Comment:
I'm struggling to understand the change.

I get that this is saving a string => byte[] conversion for the static table, but that's a one-time startup thing and could be addressed in other ways.

I get that we were previously converting the span name/value to byte[] in HeaderField, which is something I commented on in the original PR.  But there are different ways to address that, basically just making sure that the callers of this ctor are passing in logically immutable byte[]s.

Storing byte[] is in general going to result in storing half as much memory for these headers, we're reading bytes from the wire and writing bytes to the wire, so using byte[] seems like it should actually be a better format that would avoid transcoding costs.

What am I missing?

Reply:
This change is targeting the dynamic table decoder: rather than have HttpClient/Kestrel allocate new strings based on byte arrays, they are allocated just once and then reused.

Reply:
Why would they allocate strings at all?

Reply:
Oh, e.g. to hand out as part of the HttpClient response header collection?